05-dijkstras/06-dijkstras-vs-negative-graphs.md

```diff
@@ -0,0 +1,122 @@
+
+Let's walk through an example of Dijkstra’s Algorithm being implemented on a small negatively weighted graph to find the shortest path from Node A to Node D.
+
+![Dijkstra vs.Negative Graph-1](images/negative-1.png)
+
+Step 0. We start with a weighted graph. Notice that the weight on the edge connecting B to C is -100. The graph is directed however this is arbitrary.
+
+![Dijkstra vs.Negative Graph-2](images/negative-2.png)
+
+Step 1. We identify our start node, Node A, and set the cost to travel from the start node to all other nodes to infinity. The cost to travel to Node A (the start node itself) is set to 0. Something else that will prove to be useful is a list that keeps track of the nodes we’ve already visited. Finally, we add Node A to the top of our priority queue.
+
+```py
+priority queue:
+	- A
+
+visited:
+	
+
+costs/distances:
+	- A: 0 
+    - B: ∞ 
+    - C: ∞ 
+    - D: ∞
+```
+
+![Dijkstra vs.Negative Graph-3](images/negative-3.png)
+
+Step 2. Given that Node A is the only node to have a finite distance it moves to the top of our queue and we begin looking at the edges of Node A’s neighboring nodes. Here we check to see if the cost of visiting each unvisited neighbor node from the current node we are visiting is less than the estimated cost we previously noted. Here we can see that 2 and 1, for B and D respectively, are less than infinity. Therefore we will update the distances accordingly. Finally, we add our start node to the list of visited nodes so we don’t visit it again. 
+
+```py
+current: A 
+
+visited nodes: 
+    - A 
+
+priority queue: 
+ 	- A
+	- D
+	- B
+
+costs/distances: 
+    - A: 0 
+    - B: 2
+    - C: ∞ 
+    - D: 1
+```
+![Dijkstra vs.Negative Graph-4](images/negative-4.png)
+
+Step 3. The next step is to visit the node that is the closest to our start node, which is Node D due to the cost of visiting being 1, whereas Node B has a cost of 2. Here we would look at the cost of visiting all of Node D’s unvisited neighboring nodes. However, since Node D’s only neighbor is Node A, a node we’ve already visited, there are no costs to consider. 
```

Comment:
Consider splitting into two shorter sentences. 
"look at" → "compute"

```suggestion
Step 3. The next step is to visit the node that is the closest to our start node, which is Node D. Node D has a cost of 1, whereas Node B has a cost of 2. Here we would compute the cost of visiting all of Node D’s unvisited neighboring nodes. However, since Node D’s only neighbor is Node A, a node we’ve already visited, there are no costs to consider. 
```